chroma_api/stratagem.py

```diff
@@ -0,0 +1,145 @@
+from django.db import models
+from django.db.models import Q
+from django.core.exceptions import ObjectDoesNotExist
+from toolz.functoolz import pipe
+
+import tastypie.http as http
+
+from tastypie import fields
+from tastypie.resources import Resource
+from chroma_api.authentication import AnonymousAuthentication
+from chroma_core.services.job_scheduler.job_scheduler_client import JobSchedulerClient
+from tastypie.authorization import DjangoAuthorization
+from tastypie.validation import Validation
+from chroma_api.validation_utils import validate
+from chroma_api.utils import custom_response, dehydrate_command
+from chroma_core.models import (
+    StratagemConfiguration,
+    ManagedHost,
+    ManagedMdt,
+    ManagedTargetMount,
+    ManagedFilesystem,
+    Command,
+)
+
+from chroma_api.chroma_model_resource import ChromaModelResource
+
+
+class RunStratagemValidation(Validation):
+    def is_valid(self, bundle, request=None):
+        purge_duration = bundle.data.get("purge_duration")
+        report_duration = bundle.data.get("report_duration")
+
+        if "filesystem" not in bundle.data:
+            return {"code": "filesystem_required", "message": "filesystem required when running stratagem."}
+        elif isinstance(purge_duration, int) and isinstance(report_duration, int) and report_duration >= purge_duration:
+            return {"code": "duration_order_error", "message": "Report duration must be less than purge duration."}
+
+        fs_identifier = str(bundle.data.get("filesystem"))
+
+        if not any(
+            map(
+                lambda x: str(x.get("id")) == fs_identifier or str(x.get("name")) == fs_identifier,
+                ManagedFilesystem.objects.values("id", "name"),
+            )
+        ):
+            return {
+                "code": "filesystem_does_not_exist",
+                "message": "Filesystem {} does not exist.".format(fs_identifier),
+            }
+
+        fs_id = filter(
+            lambda x: str(x.get("id")) == fs_identifier or str(x.get("name")) == fs_identifier,
+            ManagedFilesystem.objects.values("id", "name"),
+        )[0].get("id")
```

Comment:
This and the check about could be done with a single query (excuse poor formatting, you'll need to run black):

```python
fs_identifier = bundle.data.get("filesystem")
fs = ManagedFilesystem.objects.filter(Q(id=fs_id) | Q(name=fs_id)).values('id').first()

if fs is None:
 return {
                "code": "filesystem_does_not_exist",
                "message": "Filesystem {} does not exist.".format(fs_identifier),
            }

fs_id = fs.get(''id")
```

Reply:
I abstracted this logic out from a previous comment so I will apply the enhancement there.

Reply:
Actually, I tried using this and it fails with this:
```
ManagedFilesystem.objects.filter(Q(id=fs_identifier) | Q(name=fs_identifier))
*** ValueError: invalid literal for int() with base 10: 'fs18'
```
`fs_identifier` is 'fs18' but `id` expects an integer so it throws an exception.

Reply:
any easy way around that?

Reply:
I guess you will need to try and parse to an int,  and handle the error if parsing fails.

Reply:
Ok, i've updated:
```
def get_fs_id_from_identifier(fs_identifier):
    try:
        id = int(str(fs_identifier), 10)
        return ManagedFilesystem.objects.filter(id=id).values("id").first()
    except ValueError:
        return ManagedFilesystem.objects.filter(name=fs_identifier).values("id").first()
```